### DISCUSSION TITLE
Crosshair position in the confocal window

### DISCUSSION THREAD

**Discussion by Olinio** [Ulm-IQO/qudi-core#35] [Q&A] [answered]
Hey guys,

I am looking for a way to get the xy-position of the crosshair in the confocal gui.
In the old qudi one could get it via sth. like _confocalgui_.__mw.xy_refocus_ViewWidget.crosshair_position_.
Right now I can't find something similar in the new qudi.

I am not looking for the actual position of the scanner that you can get from the scanning probe locig via _scanner_position_.

Thanks!

**Comment by Neverhorst** [comment, marked as answer]
Hi Olinio,

do you need this information to adapt the GUI module somehow? Because one core principle of qudi is that the logic modules are the sole command center of an experiment holding all process variables exclusively. It must work without any GUI module connected.  
I'm just asking because getting the xy-position from a GUI element has a certain smell to it that should probably be avoided.

The crosshair does not contain any more information than you can get from the `ScanningProbeLogic` itself.  
The new scanning toolchain distinguishes between the target position (the ideal position) and the actual scanner position (the "real" position, _can_ be equal to the target position if no feedback sensor is present).

So in your case I would recommend getting the xy-position you are searching for from the property `ScanningProbeLogic.target_position` or by connecting to the Qt signal `ScanningProbeLogic.sigScannerTargetChanged`.

If you really want the crosshair position from the GUI widget after all:
```python
crosshair_position = scanner_gui.scan_2d_dockwidgets[('x', 'y')].scan_widget.marker_position
```
If it is not the x-y-scan you are interested in, replace `('x', 'y')` accordingly.

__But again I strongly suggest to use `scanning_probe_logic.target_position` instead.__

**Reply by Olinio** [reply]
Hey,

I was indeed looking for the target position, thanks for the help.

**Reply by Neverhorst** [reply]
Thought so. ;) Happy to help.
